src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -647,7 +647,7 @@ private static bool InterfacesAreDistinct(NamedTypeSymbol type, ConsList<Symbol>
                 return true;
             }
 
-            if (typeArgument.IsPointerType() || typeArgument.IsRestrictedType())
+            if (typeArgument.IsPointerType() || typeArgument.IsRestrictedType() || typeArgument.SpecialType == SpecialType.System_Void)
```

Comment:
It looks like there is no unit-test for this new code path. Consider adding one.